From 7c31329cc12153443eeec29bc1d2eaf755becf82 Mon Sep 17 00:00:00 2001
From: Biju Chakkachattil <bijux.chakkachattil@intel.com>
Date: Thu, 2 Apr 2015 02:42:47 +0530
Subject: [PATCH] PMIC-BCU: Reorder the vwarnb interrupt setup

The interrupt shall be enabled when the probe is terminated
to make sure that it will not access unitialized data.

Change-Id: I18201a6a41f52442974de9a4b7adcdc8d52f8132
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1318
Signed-off-by: Jean-Christophe PINCE <jean-christophe.pince@intel.com>
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/hwmon/intel_wcove_bcu.c | 18 +++++++++---------
 1 file changed, 9 insertions(+), 9 deletions(-)

diff --git a/drivers/hwmon/intel_wcove_bcu.c b/drivers/hwmon/intel_wcove_bcu.c
index a9b794f..2d13c86 100644
--- a/drivers/hwmon/intel_wcove_bcu.c
+++ b/drivers/hwmon/intel_wcove_bcu.c
@@ -522,14 +522,8 @@ static int wcove_bcu_probe(struct platform_device *pdev)
 		goto exit_free;
 	}
 
-	/* Unmask 1st level BCU interrupt in the mask register */
-	ret = intel_soc_pmic_clearb(MIRQLVL1_REG, (u8)MBCU);
-	if (ret) {
-		dev_err(&pdev->dev, "Unmasking of BCU failed:%d\n", ret);
-		goto exit_hwmon;
-	}
-
 	/* Register for Interrupt Handler */
+	INIT_DELAYED_WORK(&info->vwarnb_irq_work, wcove_bcu_vwarnb_irq_work);
 	ret = request_threaded_irq(info->irq, wcove_bcu_intr_handler,
 						wcove_bcu_intr_thread_handler,
 						IRQF_NO_SUSPEND,
@@ -551,10 +545,16 @@ static int wcove_bcu_probe(struct platform_device *pdev)
 	/* enable voltage and current trip points */
 	wcove_bcu_enable_trip_points(info);
 
-	INIT_DELAYED_WORK(&info->vwarnb_irq_work, wcove_bcu_vwarnb_irq_work);
-
 	/* Create debufs for the basincove bcu registers */
 	wcove_bcu_create_debugfs(info);
+
+	/* Unmask 1st level BCU interrupt in the mask register */
+	ret = intel_soc_pmic_clearb(MIRQLVL1_REG, (u8)MBCU);
+	if (ret) {
+		dev_err(&pdev->dev, "Unmasking of BCU failed:%d\n", ret);
+		goto exit_freeirq;
+	}
+
 	return 0;
 
 exit_freeirq:
-- 
1.9.1

